### PR TITLE
Bug #109 Fix tests and Elastica compatibility

### LINKED ISSUE #109: Deprecated to fix with the new Elastica version
See https://github.com/ruflin/Elastica/releases

> Deprecated Elastica\Reindex::WAIT_FOR_COMPLETION_FALSE, use a boolean as parameter instead by @franmomu https://github.com/ruflin/Elastica/pull/2070

### PR DESCRIPTION
**Pull request by damienalexandre**
Fix #109 and tests.

PHPStan does not report some of our errors anymore.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -80,6 +80,9 @@ jobs:
             - name: Install Symfony Flex
               run: composer global require --no-progress --no-scripts --no-plugins symfony/flex
 
+            - name: Enable Symfony Flex
+              run: composer global config --no-plugins allow-plugins.symfony/flex true
+
```
Comment (lyrixx) [inline .github/workflows/ci.yml:85]: Why? What to does it bring?

Comment (damienalexandre) [reply .github/workflows/ci.yml:85]: Please see previous build errors: https://github.com/jolicode/elastically/runs/7856878624?check_suite_focus=true#step:6:9

Comment (lyrixx) [reply .github/workflows/ci.yml:85]: Oh yes ! 
But IMHO it should goes to this repository directly.

Comment (damienalexandre) [reply .github/workflows/ci.yml:85]: Nope. Flex is only installed in the CI.

**Pull request merged**